src/Digdir.Domain.Dialogporten.Application/Common/ICompactJwsGenerator.cs

```diff
@@ -0,0 +1,147 @@
+    private static readonly JsonSerializerOptions _serializerOptions = new()
+    {
+        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
+    };
+
+    public Ed25519Generator(IOptions<ApplicationSettings> applicationSettings)
+    {
+        _applicationSettings = applicationSettings.Value;
+    }
+
+    public string GetCompactJws(Dictionary<string, object?> claims)
+    {
+        InitSigningKey();
+
+        var header = JsonSerializer.SerializeToUtf8Bytes(new
+        {
+            alg = "EdDSA",
+            typ = "JWT",
+            kid = _kid
+        });
+
+        var payload = JsonSerializer.SerializeToUtf8Bytes(claims, _serializerOptions);
+
+        var bufferSize = Base64Url.GetMaxEncodedToUtf8Length(header.Length) +
+                         Base64Url.GetMaxEncodedToUtf8Length(payload.Length) +
+                         Base64Url.GetMaxEncodedToUtf8Length(SignatureAlgorithm.Ed25519.SignatureSize) +
+                         2; // For the dots
+
+        Span<byte> buffer = stackalloc byte[bufferSize];
+
+        Base64Url.Encode(header, buffer, out var bytesWritten);
+        var dotIndex = bytesWritten;
+        buffer[dotIndex] = (byte)'.';
+
+        Base64Url.Encode(payload, buffer[(dotIndex + 1)..], out bytesWritten);
+        dotIndex += bytesWritten + 1;
+        buffer[dotIndex] = (byte)'.';
+
+        var dataToSign = Encoding.UTF8.GetString(buffer[..dotIndex]);
+        var signature = SignatureAlgorithm.Ed25519.Sign(_privateKey!, Encoding.UTF8.GetBytes(dataToSign));
+
+        Base64Url.Encode(signature, buffer[(dotIndex + 1)..], out bytesWritten);
+
+        return Encoding.UTF8.GetString(buffer[..(dotIndex + 1 + bytesWritten)]);
+    }
+
+    public bool VerifyCompactJws(string compactJws)
+    {
+        var parts = compactJws.Split('.');
+        if (parts.Length != 3) return false;
+
+        var header = Base64Url.Decode(parts[0]);
+
+        var headerJson = JsonSerializer.Deserialize<JsonElement>(header);
+        if (headerJson.TryGetProperty("kid", out var kid))
+        {
+            if (kid.GetString() != _kid) return false;
+        }
+        else
+        {
+            return false;
+        }
+
+        var signature = Base64Url.Decode(parts[2]);
+        return SignatureAlgorithm.Ed25519.Verify(_publicKey!, Encoding.UTF8.GetBytes(parts[0] + '.' + parts[1]), signature);
+    }
+
+    private void InitSigningKey()
+    {
+        if (_kid != null) return;
+
+        var keyPair = _applicationSettings.Dialogporten.Ed25519KeyPairs.Primary;
+        _kid = keyPair.Kid;
+        _privateKey = Key.Import(SignatureAlgorithm.Ed25519,
+            Base64Url.Decode(keyPair.PrivateComponent), KeyBlobFormat.RawPrivateKey);
+        _publicKey = PublicKey.Import(SignatureAlgorithm.Ed25519,
+            Base64Url.Decode(keyPair.PublicComponent), KeyBlobFormat.RawPublicKey);
+    }
+}
+
+internal sealed class LocalDevelopmentCompactJwsGeneratorDecorator : ICompactJwsGenerator
```

Comment:
Så vi trenger ikke toolen du har lagt - strengt tatt?

Reply:
Jo, toolen trenger du om du ønsker å bruke denne funksjonaliteten i det hele tatt. Decoratoren deaktiverer jo bare alt. Vi kunne sett på at decoratoren i stedet genererte opp nøkkelpar, men det synes jeg blir i overkant mye kode/logikk kun for localdev. Som nevnt over vil toolen også kunne brukes ifm oppsett av nytt miljø, og potensielt rotasjon av nøkler.